modules/claims_api/app/services/claims_api/bgs_to_lighthouse_claims_mapper_service.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class BGSToLighthouseClaimsMapperService < ClaimsApi::Service
+    attr_accessor :bgs_claims, :lighthouse_claims
+
+    def initialize(bgs_claims:, lighthouse_claims:)
+      @bgs_claims        = bgs_claims
+      @lighthouse_claims = lighthouse_claims
+    end
+
+    def process
+      return [] unless claims_exist?
+
+      mapped_claims = map_claims
+      mapped_claims = append_remaining_lighthouse_claims(mapped_claims: mapped_claims) if lighthouse_claims.present?
+      mapped_claims
+    end
+
+    private
+
+    def claims_exist?
+      return true if bgs_claims.key?(:bnft_claim_detail) || lighthouse_claims.present?
+
+      false
+    end
+
+    def map_claims
+      bgs_claims[:bnft_claim_detail].map do |bgs_claim|
+        matching_claim = find_bgs_claim_in_lighthouse_collection(claim: bgs_claim)
+        if matching_claim
+          remove_bgs_claim_from_lighthouse_collection(claim: matching_claim)
```

Comment:
Just for my own understanding, for a given veteran:

1. we're grabbing all their claims submitted through Lighthouse
2. we're grabbing all their claims that exist in BGS
3. we're removing any Lighthouse claims from the BGS results
4. we're merging the 2 groups, resulting in all unique claims found between Lighthouse and BGS

does that sound right?

Reply:
That's pretty close.

1. grab all of their claims that BGS knows about
2.  grab all of their claims that Lighthouse knows about
3. iterate the BGS collection and see if there is a claim in the Lighthouse collection with a matching external id
4. if there is a match
-- remove the matching claim from the Lighthouse collection
-- use the guid of the matching Lighthouse claim as the value for the returned `id` attribute
5. if there is not a match
-- use the `bnft_claim_id` of the BGS claim as the value for the returned `id` attribute
6. after iterating the BGS collection, if there are any claims remaining in the Lighthouse collection AND they don't have a status of `established`, then also include them in the collection that is being returned.

You might look at the specs I added to see if that helps.  I tried to keep the specs really small to make it easier to follow.